### DISCUSSION TITLE
Adding auto-import type recognition for threejs

### DISCUSSION THREAD

**Discussion by cesarecaoduro** [nuxt/nuxt#16792] [Questions]
Good afternoon community!
I am trying to build a 3D webgl using Nuxt3 and Threejs. I have managed to create few state and auto-import them from composable.
```js
import { Camera, Scene, WebGLRenderer } from "three";

export const useScene = () => useState<Scene>('scene', () => new Scene());
export const useCamera = () => useState<Camera>('camera', () => new Camera());
export const useRenderer = () => useState<WebGLRenderer>('renderer', () => new WebGLRenderer());
```
The problem I am facing now, is that I can't have typecheking when I start using threejs classes inside other components

```html
// app.vue

<script setup lang="ts">

const scene : Scene = useScene();

</script>
<template>
</template>
```

The type ```Scene``` is not recognized, unless I explicitly import that in the component.
```html
// app.vue

<script setup lang="ts">
import {Scene} from 'three';
const scene : Scene = useScene();

</script>
<template>
</template>
```

Is there a better way to do it? This is my ```package.json```

```json
{
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview"
  },
  "devDependencies": {
    "@types/three": "^0.143.2",
    "nuxt": "3.0.0-rc.8"
  },
  "dependencies": {
    "three": "^0.143.0"
  }
}
```
anything else, is just vanilla Nuxt3. Thanks!

**Comment by sevillaarvin** [comment]
Hmm, according to this the types should be generated automatically:

https://v3.nuxtjs.org/guide/concepts/typescript/#auto-generated-types

> Some of the references in the file are to files that are only generated within your buildDir (.nuxt) and therefore for full typings, you will need to run nuxi dev or nuxi build.

Perhaps try running dev or build, see if it works?
